open-craft/edx-platform
# Cherry-pick updates from upstream juniper.master branch

This cherry-picks security updates and fixes from [open-release/juniper.master](https://github.com/edx/edx-platform/commits/open-release/juniper.master).

**Sandbox:**

- Ocim: https://manage.opencraft.com/instance/21770/
- LMS: https://juniper-security-fixes.opencraft.hosting
- Studio: https://studio.juniper-security-fixes.opencraft.hosting

**Test instructions:**

Verify that things work correctly with the security fixes & updates applied.
The main changes are in password reset handling and certificate URLs.
- Make sure that password reset still works
- Make sure that certificate work with the new URL. To test this I set up [a certificate on the sandbox](https://studio.juniper-security-fixes.opencraft.hosting/certificates/course-v1:edX+DemoX+Demo_Course) and verified that:
  * the new URL (without the user ID) works: https://studio.juniper-security-fixes.opencraft.hosting/certificates/course-v1:edX+DemoX+Demo_Course
  * the old URL shows a warning and instructions on how to obtain a new link: https://juniper-security-fixes.opencraft.hosting/certificates/user/6/course/course-v1:edX+DemoX+Demo_Course?preview=honor

**Reviewers**:
